integrated_channels/integrated_channel/learner_data.py

```diff
@@ -134,6 +133,52 @@ def collect_learner_data(self):
                 is_passing=is_passing,
             )
 
+    def _should_skip_enrollments_data_collection(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection of the enterprise enrollment due to any configuration reason.
+
+        Reasons for skipping data collection may include data sharing consent not being granted,
+        or audit track data reporting being disabled for audit enrollments, etc..
+
+        :param enterprise_enrollment: The enterprise enrollment whose configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        return any([
+            not enterprise_enrollment.consent_available,
+            self._disabled_audit_reporting(enterprise_enrollment),
+        ])
+
+    def _disabled_audit_reporting(self, enterprise_enrollment):
+        """
+        Specifies whether to skip data collection due to audit track data reporting not being enabled.
+
+        If it is enabled, we simply skip the check, regardless of the ``enterprise_enrollment``.
+        If it isn't enabled, we check to see if the course mode is of an audit type.
+
+        :param enterprise_enrollment: The enterprise enrollment whose reporting configuration is to be checked against.
+        :return: True if we should skip, False otherwise.
+        """
+        if not self.enterprise_customer.enables_audit_data_reporting:
+            return self._is_audit_enrollment(enterprise_enrollment)
+
+        # Since audit data reporting is enabled, we always return False here.
+        return False
+
+    def _is_audit_enrollment(self, enterprise_enrollment):
```

Comment:
See comment above about the fact that this would couple it to the course enrollment API. What do you think about that?